src/rgw/rgw_swift_auth.cc

```diff
@@ -524,8 +524,10 @@ void RGW_SWIFT_Auth_Get::execute()
       protocol = "https";
     } else {
       server_port = s->info.env->get("SERVER_PORT");
-      add_port = (strcmp(server_port, "80") != 0);
-      protocol = "http";
+      if (server_port) {
```

Comment:
could you prefix the title of your commit message with the subcomponent your are changing ? see https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes, in this case, it would be "rgw: "